packages/node-runtime-worker-thread/src/rpc.ts

```diff
@@ -171,14 +171,21 @@ export type Caller<
 
 export function createCaller<Impl extends {}>(
   methodNames: Extract<keyof Impl, string>[],
-  messageBus: RPCMessageBus
+  messageBus: RPCMessageBus,
+  processors: Partial<
+    Record<typeof methodNames[number], (...input: any[]) => any[]>
+  > = {}
 ): Caller<Impl, typeof methodNames[number]> {
   const obj = {};
   const inflight = new Set<CancelablePromise<unknown>>();
   methodNames.forEach((name) => {
     const c = caller(name as string, getRPCOptions(messageBus));
     (obj as any)[name] = async(...args: unknown[]) => {
-      const promise = c(...args);
+      const processed =
+        typeof processors[name] === 'function'
+          ? processors[name]?.(...args)
```

Comment:
Sucks that we already checked that processors[name] is a function and therefore it must exist, but then typescript complains so I do ?. which then causes knock-on effects..